nixos/modules/services/networking/syncthing.nix

```diff
@@ -621,6 +621,21 @@ in {
 
   config = mkIf cfg.enable {
 
+    vars.generators.syncthing = {
```

Comment:
I'm not understanding the glue between this generator and the syncthing service. How do the files created by this generator actually get wired up into the running service?

Reply:
Hope I am not misreading your question, but the pem files are seemingly simply used to set conventional nixos options such as on [line 639-641](https://github.com/NixOS/nixpkgs/pull/370444/files#diff-fd4c91fd302760b673b3e9b6c8d3474f0ae371005b6be1baff552d0aa7ffe223R639-R641) in this example.

The device id in `syncthing.pub` does not seem to be used in the example, but only stored. Presumably for outside use just as scripted setups of other devices to sync with.